CryptoTechReminderSystem/Gateway/SlackGateway.cs

```diff
@@ -48,15 +50,30 @@ private async Task<object> PostChatMessage(StringContent content)
 
         public IList<SlackDeveloper> RetrieveDevelopers()
         {
-            return new List<SlackDeveloper>();
-        }
-    }
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _token);
 
-    public class PostMessageRequest
-    {
-        [JsonProperty("channel")]
-        public string Channel { get; set; }
-        [JsonProperty("text")]
-        public string Text { get; set; }
+            var result = GetUsers().Result;
+            var users = result["members"];
+
+            return users.Select(developer => new SlackDeveloper
+                {
+                    Id = developer["id"].ToString()
```

Comment:
I think this request should return the email address along with the id? Or how are we going to match them with the harvest records?

Reply:
You're right! Will add that now.

Reply:
Strongly agree with Csaba.

Reply:
I've added another unit test `CanGetEmailOfSlackDeveloper()` and ensured the email is returned as part of the `SlackGateway` domain object. https://github.com/chubberlisk/crypto_tech/pull/12/commits/5f84738fbd9c01f8099cba3b26f24fa62fb1fa7d 

https://github.com/chubberlisk/crypto_tech/blob/5f84738fbd9c01f8099cba3b26f24fa62fb1fa7d/CryptoTechReminderSystem/Gateway/SlackGateway.cs#L61

https://github.com/chubberlisk/crypto_tech/blob/5f84738fbd9c01f8099cba3b26f24fa62fb1fa7d/CryptoTechReminderSystem/DomainObject/SlackDeveloper.cs#L6